tensorflow/contrib/tensorrt/resources/trt_allocator.cc

```diff
@@ -37,8 +37,22 @@ void TRTCudaAllocator::free(void* memory) { cudaFree(memory); }
 
 void* TRTDeviceAllocator::allocate(uint64_t size, uint64_t alignment,
                                    uint32_t flags) {
+  // WAR for allocator alignment requirement
+  alignment = 512;
   assert((alignment & (alignment - 1)) == 0);  // zero or a power of 2.
-  void* mem = allocator_->AllocateRaw(alignment, size);
+  void* mem = allocator_->AllocateRaw(alignment, size + alignment);
+
+  CHECK(mem_pool.count(mem) == 0);
+  mem_pool.insert(mem);
+  CHECK(mem);
+  void* alloc_mem = mem;
+  uint64_t total_size = size + alignment;
+  std::align(alignment, size, mem, total_size);
```

Comment:
Do we need to do `mem = std::align(blablabla)`? According to https://en.cppreference.com/w/cpp/memory/align, the returned value is `The adjusted value of ptr, or null pointer value if the space provided is too small`. So I think we need to check the returned result?